http2.py

```diff
@@ -0,0 +1,122 @@
+import socket
+import sys
+import mimetypes
+import os
+from os import getcwd
+from os.path import isdir
+from os.path import isfile
+from os import listdir
+
+
+_root = os.path.join(os.getcwd(), 'webroot')
+
+
+class InvalidHTTPCodeError(Exception):
+    pass
+
+
+class NotGETRequestError(Exception):
+    pass
+
+
+class NotHTTP1_1Error(Exception):
+    pass
+
+
+class BadRequestError(Exception):
+    pass
+
+
+class ResourceNotFound(Exception):
+    pass
+
+
+class HttpServer(object):
+    def http_server():
+        user_socket = socket.socket(
+            socket.AF_INET,
+            socket.SOCK_STREAM,
+            socket.IPPROTO_IP)
+        user_socket.bind(('127.0.0.1', 50000))
+        user_socket.listen(1)
+
+        while True:
+            conn, addr = user_socket.accept()
+            message = ''
+
+            while True:
+                msg_recieved = conn.recv(32)
+                message += msg_recieved
+                if len(msg_recieved) < 32:
+                    break
+            try:
+                # This seems to be a mac-specific bug
+                conn.shutdown(socket.SHUT_RD)
+            except Exception, e:
+                pass
+            print message
+            conn.sendall(message)
+            conn.shutdown(socket.SHUT_WR)
+            conn.close()
```

Comment:
Again, you aren't yet using any of the functions you wrote below in the server block here.  You want to do that.  Otherwise this isn't an HTTP server.